src/query/api/v1/handler/namespace/common.go

```diff
@@ -190,6 +191,16 @@ func validateNamespaceAggregationOptions(mds []namespace.Metadata) error {
 				resolutionRetentionMap[key] = true
 			}
 		}
+
+		indexOpts := md.Options().IndexOptions()
+		indexBlockSize := indexOpts.BlockSize()
+		retentionBlockSize := retentionOpts.BlockSize()
+		if indexBlockSize != retentionBlockSize {
```

Comment:
Would it make more sense to put this logic on the `md.Options().Validate()` method?

Reply:
I'll look to see if this is better. I think we might want to only enforce this on-creation though so we don't break for any existing ns that happen to have this configuration

Reply:
Ok yeah actually I think we actually might have to put in Validate since that gets run on the incoming settings on-create, while the place I put the enforcement happens on existing ones as well which I think we don't want. Making this change.